### PR TITLE
Add support for .eslintignore

### PR DESCRIPTION
**Pull request by danielsousaio**
Hello there! Thanks for creating this action.

I've noticed that `.eslintignore` or `ignoredPaths` are not respected since we are providing the file paths as an argument to the [`eslint` command call](https://github.com/sibiraj-s/action-eslint/blob/1e1b44c70776726531fce03718ef3af9979d58fe/src/eslint.ts#L43) directly.

This PR reads from `.eslintignore` and filters the changed files against it, effectively allowing for using it in parallel with the action. This can prove useful for repositories where there's `dist` directories checked in that you don't want to be linted.

### CODE REVIEW

**Comment by sibiraj-s** [comment]
Thanks. Adding default support for `.eslintingore` is good. I do have few questions though

1. Is it not working with` eslint-args: '--ignore-path=.eslintignore --quiet'`?
2. Should we consider [bin-path](https://github.com/sibiraj-s/action-eslint/blob/master/action.yml#L30) while looking for eslintignore. now it only looks at the root.
3. Should eslintignore be considered when custom ignore-path is configued `--ignore-path=.eslintignore` 

> I've noticed that .eslintignore or ignoredPaths

Not sure what you are referring as ignoredPath. Are you referring to the one configured in eslintignore?

**Comment by danielsousaio** [comment]
> Is it not working with eslint-args: '--ignore-path=.eslintignore --quiet'?

Ignoring the path works, but an interesting side effect occurs, it tries to load more configs due to the way ESLint works.
https://github.com/crowdbotics/modules/pull/394

That PR demonstrates the issue, the logs show that a nested configuration file was picked up. Our repository contains a React Native app template and we don't want to lint the template itself, and we don't want to cascade into the template's ESLint configuration file. In this specific case using that nested config is invalid and in fact there's a missing dependency error in the GH Action logs, as expected. That is the reason why I went with filtering files directly using `.eslintignore` so that their paths are not even fed as args to the command call.

> Should we consider [bin-path](https://github.com/sibiraj-s/action-eslint/blob/master/action.yml#L30) while looking for eslintignore. now it only looks at the root.

What I could understand from the documentation, is that only one `.eslintignore` file is ever used. There's no cascading for those, only for the configuration files:
> Only one .eslintignore file can be used at a time, so .eslintignore files other than the one in the current working directory will not be used.

https://eslint.org/docs/latest/user-guide/configuring/ignoring-code

> Should eslintignore be considered when custom ignore-path is configued --ignore-path=.eslintignore

For our use case it should since otherwise configuration cascading can cause issues. It only cascades through paths that it lints, and if I don't pass those paths as args then it doesn't cascade there.

> Not sure what you are referring as ignoredPath. Are you referring to the one configured in eslintignore?

Sorry, that was a typo, I meant `ignoredPatterns` in `.eslintrc.json`/`.eslintrc.js`.

Thanks for your time! Let me know what you think, maybe this is just too specific for our use case and if so that's okay and we can keep the fork going.

**Comment by sibiraj-s** [comment]
Hi, I am really sorry for the no reply. Got occupied with my day job and I totally forgot about this. Busy weeks.

Thanks for the detailed explanation. I understand the problem. One thing I don't understand is this

> Sorry, that was a typo, I meant ignoredPatterns in .eslintrc.json/.eslintrc.js.

Why this is not working for you 🤔 . I've been personally using this and even this repo uses it. See 

https://github.com/sibiraj-s/action-eslint/blob/1e1b44c70776726531fce03718ef3af9979d58fe/.eslintrc.yml#L10-L12

**Comment by sibiraj-s** [comment]
Hi @danielsousaio Can you resolve the conflicts. I'll merge. We can resolve the concerns later. I tried to resolve it, but I couldn't push to your PR since it is raised from fork's master.

**Comment by danielsousaio** [comment]
> Hi @danielsousaio Can you resolve the conflicts. I'll merge. We can resolve the concerns later. I tried to resolve it, but I couldn't push to your PR since it is raised from fork's master.

Hey there, sure! Done

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "eslint": "^8.23.0",
     "eslint-config-pegasus": "^3.4.1",
     "husky": "^8.0.1",
+    "ignore": "^5.2.0",
```
Comment (sibiraj-s) [inline package.json:43]: Should be a dependency. I can fix this after merge.

**Review by sibiraj-s — APPROVED** [review]

**Pull request merged**